olps/strategies/bah_strategy.py

```diff
@@ -0,0 +1,41 @@
+from .strategy import Strategy
+from ..datasources.datasource import DataSource
+import numpy as np
+
+class BAHStrategy(Strategy):
+    # TODO implement
+    def update_weights(self, market_data: DataSource) -> None:
+        if len(market_data.price_relatives.shape) == 1:
+            prv = market_data.price_relatives
+        else:
+            prv = market_data.price_relatives[:, -1]
+        wealth = np.dot(prv, self.weights)
+        print('wealth', wealth)
+        self.weights = self.weights * prv
+        self.weights = self.weights / sum(self.weights)
+        y = self.update_wealth(wealth)
+        print('y', y)
+        print('cumu', self.cumulative_wealth)
+
+def main():
```

Comment:
Regarding tests: I think it would be great if we could add more explicit tests for each strategy in the `tests/` folder. For an example, see `test_strategy.py`, where we're actually comparing if calculated numerical values are close to what we expect.

Additionally, I think removing the debugging code would be nice.